### PR TITLE
Handle a few edge cases in the book hotel flow

### PR DESCRIPTION
**Pull request by ArjaanBuijk**
- when trip_destination is not in the database as a hotel location, give a clear message what locations are available.
- handle the user declining a hotel_booking_confirmation

### CODE REVIEW

File: calm_llm/data/flows/book_hotel.yml [hunk 1]
```diff
@@ -4,7 +4,16 @@ flows:
     steps:
       - collect: trip_destination
         reset_after_flow_ends: false      
-      - collect: hotel_price_range
+      - id: "verify_hotel_location_is_in_db"
```
Comment (amn41) [inline calm_llm/data/flows/book_hotel.yml:7]: wouldn't it be simpler to use a rejection? https://rasa.com/docs/rasa-pro/building-assistants/validating-slot-values#validating-a-slot-values-format-with-a-condition

Comment (ArjaanBuijk) [reply calm_llm/data/flows/book_hotel.yml:7]: Yeah, good question. I thought about that but decided that the current approach follows best practices, for following reasons:

- I considered a `rejection` but the check is too difficult to do it easily via pyred. It needs to handle conversion to lowercase and then check against a range of values. Is probably possible, but I think it will be hard to maintain and debug.

- I considered a slot_validation with a custom action on trip_destination, but that would hide the logic inside a custom action. The trip_destination slot is a global slot, and to check if it is in the database as a hotel location needs to run only for the search_hotel flow. We should not be hiding that type of logic inside a custom action.

File: calm_llm/data/flows/book_hotel.yml [hunk 2]
```diff
@@ -27,4 +36,12 @@ flows:
         description: Name of the hotel which the user has selected. Must be an exact element of the slot "hotel_search_results_readable". If the user wants an option that isn't listed, do not fill the slot with that value.
       - collect: hotel_booking_confirmation
         ask_before_filling: true
-      - action: utter_confirm_booked_hotel
+      - noop: true
+        next:
```
[the remark this answers is not in the document]
Comment (ArjaanBuijk) [reply calm_llm/data/flows/book_hotel.yml:40]: Yes, it can be. 
The reason I used this structure to be consistent with the structure that was used after the `collect: hotel_end_date` in the same flow.

I will change both to skip the `noop: true`

File: calm_llm/domain/book_hotel.yml [hunk 3]
```diff
@@ -28,13 +33,23 @@ slots:
...
-    - text: "I've found some great options for your stay: {hotel_search_results_readable}"
+    - text: |
+        I've found some great options for your stay: 
+        {hotel_search_results_readable}
   utter_ask_hotel_booking_confirmation:
     - text: "Excellent choice! Just to confirm, you'd like me to book the {hotel_name} for your stay from {hotel_start_date_readable} to {hotel_end_date_readable}, correct?"
   utter_confirm_booked_hotel:
     - text: Great news! I've successfully booked your stay at the {hotel_name}
+  utter_will_not_book_hotel:
+    - text: Ok, I will not book anything.
```
[the remark this answers is not in the document]
Comment (Balowen) [reply calm_llm/domain/book_hotel.yml]: Won't the pattern_completed run the utter_else resulting in asking "Is there something else I can do for you?" twice?

[the remark this answers is not in the document]
Comment (amn41) [reply calm_llm/domain/book_hotel.yml]: we disabled that pattern in this bot, because there are a lot of multi-skill conversations so it's a bit tedious if you keep getting asked that!

File: calm_llm/data/flows/book_hotel.yml (hunk 1 above)
Comment (amn41) [reply calm_llm/data/flows/book_hotel.yml:7]: > We should not be hiding that type of logic inside a custom action

Agreed! This also makes it easier to work in Studio, because the logic is in the flow

File: calm_llm/domain/book_hotel.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Balowen) [reply calm_llm/domain/book_hotel.yml]: Ah. makes sense. Thanks!

File: calm_llm/data/flows/book_hotel.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ArjaanBuijk) [reply calm_llm/data/flows/book_hotel.yml:38]: This is now fixed

**Comment by ArjaanBuijk** [comment]
@amn41 , Can you have another look?
- I removed the noop in two places. That nicely cleaned things up
- I added two tests that verify the edge cases handled by these updates
